### ISSUE TITLE
False positive when using AsyncIterator

### ISSUE THREAD

**Issue by maltevesper** [microsoft/pyright#6598]
Pyright (pyright-1.1.338) falsely warns on the implementation of `bar()`, while `baz()` does not generate any warnings.

```python
from typing import AsyncIterator


async def foo() -> AsyncIterator[int]:
    yield 1


async def bar() -> int:
    return await anext(foo()) # pyright warns here


async def baz() -> int:
    temp = await anext(foo())
    return temp
```

Pyright output:

> .py:9:24 - error: Argument of type "AsyncIterator[int]" cannot be assigned to parameter "__i" of type "_SupportsSynchronousAnext[_AwaitableT@anext]" in function "anext"
>     "AsyncIterator[int]" is incompatible with protocol "_SupportsSynchronousAnext[Coroutine[Any, Any, int]]"
>       "__anext__" is an incompatible type
>         Type "() -> Awaitable[int]" cannot be assigned to type "() -> _AwaitableT_co@_SupportsSynchronousAnext"
>           Function return type "Awaitable[int]" is incompatible with type "_AwaitableT_co@_SupportsSynchronousAnext"
>             Type "Awaitable[int]" cannot be assigned to type "Coroutine[Any, Any, int]" (reportGeneralTypeIssues)
> 1 error, 0 warnings, 0 informations

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.339, which I just published. It will also be included in a future release of pylance.
